tests/admin_changelist/tests.py

```diff
@@ -364,6 +366,29 @@ def test_result_list_html(self):
             table_output,
         )
 
+    def test_action_checkbox_calls_dunder_html(self):
+        """
+        Regression test for #35606: obj repr should be used instead of
+        obj itself so __html__ is not returned.
+        """
+        Book.objects.create(title="The Book")
+        request = self.factory.get("/book/")
+        request.user = self.superuser
+        m = BookAdmin(Book, custom_site)
+        cl = m.get_changelist_instance(request)
+        cl.formset = None
+        template = Template(
+            "{% load admin_list %}{% spaceless %}{% result_list cl %}{% endspaceless %}"
+        )
+        context = Context({"cl": cl, "opts": Book._meta})
+        table_output = template.render(context)
+        self.assertNotInHTML(
```

Comment:
+1 to adding a `__str__` method
There is a pattern using `build_tbody_html` to create the row html and checking it exsists in the table that we might want to reuse (this is from `test_result_list_html`)
```python
       row_html = build_tbody_html(
            new_child,
            link,
            "name",
            '<td class="field-parent nowrap">%s</td>' % new_parent,
        )
        self.assertNotEqual(
            table_output.find(row_html),
            -1,
            "Failed to find expected row element: %s" % table_output,
        )
```
If we were to add a `__html__` method to `Child`, all of these tests which look for html built in `build_tbody_html` fail (so adding `__html__` and `__str__` to `Child` is an option, but it might be nice to do this to another model and have a separate test case 🤔)

Reply:
Thank you both for the reviews. Regarding the choice between adding `__html__` to the `Child` model or another model, I opted for the latter. This way, test failures point to one specific test instead of five. If you believe the former option is better, please let me know.